x-pack/plugins/infra/public/hooks/use_link_props.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useMemo } from 'react';
+import { stringify } from 'query-string';
+import url from 'url';
+import { url as urlUtils } from '../../../../../src/plugins/kibana_utils/public';
+import { usePrefixPathWithBasepath } from './use_prefix_path_with_basepath';
+import { useHistory } from '../utils/history_context';
+
+type Search = Record<string, string | string[]>;
+
+export interface LinkDescriptor {
+  // When an app isn't provided (for external linking) the history instance will
+  // be used to ensure either metrics or logs is used.
+  app?: string;
+  pathname?: string;
+  hash?: string;
+  search?: Search;
+}
```

Comment:
Yeah, so this is a very fair criticism. I'll look into this now, but as far as I know there isn't a way to retreive the `basename` of the `history` instance after the fact. And I don't really want to manually inspect the `location` for `metrics` or `logs`. But what could be nice is manually setting the current `app` on the history context at startup / mount, then this can just be checked for equality in `useLinkProps`. Either way, I'll improve this scenario.

Reply:
Explicitly setting the `app` on the context sounds reasonable. Alternatively, could it work to just call `createHref` with the given `app` and compare the resulting `pathname` to the value that the prefixer returns? Essentially treat it like an external link first and then optimize in case it turns out it's not external after all?

Reply:
Yeah, that could work. I suppose there's the small overhead of running both when really only one is needed, but they're pretty cheap functions.